crates/rattler_index/src/lib.rs

```diff
@@ -0,0 +1,197 @@
+        if path.as_os_str().eq("info/index.json") {
+            return package_record_from_index_json(file, &mut entry);
+        }
+    }
+    Err(std::io::Error::new(
+        std::io::ErrorKind::Other,
+        "No index.json found",
+    ))
+}
+
+/// Create a new `repodata.json` for all packages in the given output folder. If `target_platform` is
+/// `Some`, only that specific subdir is indexed. Otherwise indexes all subdirs and creates a
+/// `repodata.json` for each.
+pub fn index(
+    output_folder: &Path,
+    target_platform: Option<&Platform>,
+) -> Result<(), std::io::Error> {
+    let entries = WalkDir::new(output_folder).into_iter();
+    let entries: Vec<(PathBuf, ArchiveType)> = entries
+        .filter_entry(|e| e.depth() <= 2)
+        .filter_map(|e| e.ok())
+        .filter_map(|e| {
+            ArchiveType::split_str(e.path().to_string_lossy().as_ref())
+                .map(|(p, t)| (PathBuf::from(format!("{}{}", p, t.extension())), t))
+        })
+        .collect();
+
+    // find all subdirs
+    let mut platforms = entries
+        .iter()
+        .filter_map(|(p, _)| {
+            p.parent()
+                .and_then(|parent| parent.file_name())
+                .and_then(|file_name| {
+                    let name = file_name.to_string_lossy().to_string();
+                    if name != "src_cache" {
+                        Some(name)
+                    } else {
+                        None
+                    }
+                })
+        })
+        .collect::<std::collections::HashSet<_>>();
+
+    // Always create noarch subdir
+    if !output_folder.join("noarch").exists() {
+        std::fs::create_dir(output_folder.join("noarch"))?;
+        platforms.insert("noarch".to_string());
+    }
+
+    // Create target platform dir if needed
+    if let Some(target_platform) = target_platform {
+        let platform_str = target_platform.to_string();
+        if !output_folder.join(&platform_str).exists() {
+            std::fs::create_dir(output_folder.join(&platform_str))?;
+            platforms.insert(platform_str);
+        }
+    }
+
+    for platform in platforms {
+        if let Some(target_platform) = target_platform {
+            if platform != target_platform.to_string() {
+                if platform != "noarch" {
+                    continue;
+                } else {
+                    // check that noarch is already indexed if it is not the target platform
+                    if output_folder.join("noarch/repodata.json").exists() {
+                        continue;
+                    }
+                }
+            }
+        }
+
+        let mut repodata = RepoData {
+            info: Some(ChannelInfo {
+                subdir: platform.clone(),
+                base_url: None,
+            }),
+            packages: Default::default(),
+            conda_packages: Default::default(),
+            removed: Default::default(),
+            version: Some(1),
```

Comment:
```suggestion
            version: Some(2),
```

This can be `2`. We implemented https://github.com/conda-incubator/ceps/pull/61 which bumps the version to 2. I agree that this would be better implemented into the writing itself. ;)